### DISCUSSION TITLE
How to create and understand custom connectors?

### DISCUSSION THREAD

**Discussion by kkacstudent** [prestodb/presto#22993] [General]
I am having a very hard time understanding about how to create connectors. The sample guide for presto-example-http is not very useful. Any guide on creating and testing connectors in detail?

**Comment by ZacBlanco** [comment]
I'm not aware of any other existing guides for creating a custom connector. I think the best way to understand is by looking at examples - such as the HTTP connector. If you're struggling to understand something in the example connector, feel free to ask a more specific question about it.

At a high level though, all connectors implement portions of Presto's `Plugin` interface from the SPI: https://github.com/prestodb/presto/blob/1fda1368b1f16dc5e05f79640cc9ce837caf077d/presto-spi/src/main/java/com/facebook/presto/spi/Plugin.java#L42-L45

You can consider that the main entrypoint. Everything else in the connector revolves around the implementation for the `ConnectorFactory` specified in the plugin implementation. There are other interfaces within the SPI which are provided by methods within the `ConnectorFactory`. Those must also created to properly implement each connector. It is a pretty long chain of dependencies which can be confusing. Please let us know which parts specifically are difficult for you to understand and we can try to clarify them in the existing guide.
